lib/jekyll/commands/serve.rb

```diff
@@ -172,31 +283,51 @@ def enable_logging(opts)
         # forget to add one of the certificates.
 
         private
-        # rubocop:disable Metrics/AbcSize
         def enable_ssl(opts)
-          return if !opts[:JekyllOptions]["ssl_cert"] && !opts[:JekyllOptions]["ssl_key"]
-          if !opts[:JekyllOptions]["ssl_cert"] || !opts[:JekyllOptions]["ssl_key"]
+          jekyll_opts = opts[:JekyllOptions]
+          return if !jekyll_opts["ssl_cert"] && !jekyll_opts["ssl_key"]
+          if !jekyll_opts["ssl_cert"] || !jekyll_opts["ssl_key"]
             # rubocop:disable Style/RedundantException
             raise RuntimeError, "--ssl-cert or --ssl-key missing."
           end
+
+          Jekyll.logger.info "LiveReload:", "Serving over SSL/TLS.  If you are using a "\
+            "certificate signed by an unknown CA, you will need to add an exception "\
+            "for both #{jekyll_opts["host"]}:#{jekyll_opts["port"]} and "\
+            "#{jekyll_opts["host"]}:#{jekyll_opts["livereload_port"]}"
+
           require "openssl"
           require "webrick/https"
-          source_key = Jekyll.sanitized_path(opts[:JekyllOptions]["source"], \
-                    opts[:JekyllOptions]["ssl_key" ])
-          source_certificate = Jekyll.sanitized_path(opts[:JekyllOptions]["source"], \
-                    opts[:JekyllOptions]["ssl_cert"])
+          source_key = Jekyll.sanitized_path(
+            jekyll_opts["source"],
+            jekyll_opts["ssl_key"]
+          )
+          source_certificate = Jekyll.sanitized_path(
+            jekyll_opts["source"],
+            jekyll_opts["ssl_cert"]
+          )
           opts[:SSLCertificate] =
             OpenSSL::X509::Certificate.new(File.read(source_certificate))
           opts[:SSLPrivateKey ] = OpenSSL::PKey::RSA.new(File.read(source_key))
           opts[:SSLEnable] = true
         end
 
         private
-
         def start_callback(detached)
           unless detached
             proc do
-              Jekyll.logger.info("Server running...", "press ctrl-c to stop.")
+              @running << "."
```

Comment:
I'm using it as a mechanism to block the test from running until the server(s) can successfully start.  The only place it's actually waiting for the queue to be populated is in `test_commands_serve.rb`.  The test starts a `Serve` thread and `Serve` starts a WEBrick thread and joins it so there's not another good way I could think of to know whether WEBrick is actually up and running.

I think Queue is the appropriate choice here; the docs say "[The Queue] class provides a way to synchronize communication between threads."

That being said, I do think that it's kind of crummy to have cruft in there that's only really used for the tests.  I'm totally open to suggestions on a better way to handle this situation.